### DISCUSSION TITLE
temp变量为什么会跟着改变？

### DISCUSSION THREAD

**Discussion by Exiurs** [physics-data/faq#46] [Q&A] [answered]
permutation=dict()
        temp=generate_permutation(n-1)
        for i in range(math.factorial(n)):
            permutation[i+1]=temp[(i+n)//n]
            permutation[i+1].insert(i%n,n)
            print("temp=",temp,"permutation=",permutation)

在运行过程中，temp[(i+n)//n]会随着permutation[i+1]一起改变，是因为他们指向同一个列表吗？
还有insert会给字典里的所有列表都生效吗？？

**Comment by Exiurs** [comment, marked as answer]
赋值的时候用a=b的话，a、b为别名，指向同一个变量；
应该用a=b.copy()之类的

**Reply by heroxbd** [reply]
是这样的。
